{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f9fee31",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mNextflow 25.04.8 is available - Please consider updating your version to it\u001b[m\n",
      "\n",
      "\u001b[1m\u001b[38;5;232m\u001b[48;5;43m N E X T F L O W \u001b[0;2m  ~  \u001b[mversion 25.04.7\u001b[m\n",
      "\u001b[K\n",
      "Launching\u001b[35m `./main.nf` \u001b[0;2m[\u001b[0;1;36mnostalgic_swanson\u001b[0;2m] DSL2 - \u001b[36mrevision: \u001b[0;36m8d75f29691\u001b[m\n",
      "\u001b[K\n",
      "\n",
      "------------------------------------------------------\n",
      "                                        ,--./,-.\n",
      "        ___     __   __   __   ___     /,-._.--~'\n",
      "  |\\ | |__  __ /  ` /  \\ |__) |__         }  {\n",
      "  | \\| |       \\__, \\__/ |  \\ |___     \\`-._,-`-,\n",
      "                                        `._,._,'\n",
      "  nf-core/leoniechrissi 1.0.0dev\n",
      "------------------------------------------------------\n",
      "\u001b[1mInput/output options\u001b[0m\n",
      "  \u001b[0;34minput              : \u001b[0;32m./assets/samplesheet_copy.csv\u001b[0m\n",
      "  \u001b[0;34moutdir             : \u001b[0;32mresults/\u001b[0m\n",
      "\n",
      "\u001b[1mReference genome options\u001b[0m\n",
      "  \u001b[0;34mgenome             : \u001b[0;32mR64-1-1\u001b[0m\n",
      "  \u001b[0;34mfasta              : \u001b[0;32ms3://ngi-igenomes/igenomes//Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/WholeGenomeFasta/genome.fa\u001b[0m\n",
      "\n",
      "\u001b[1mGeneric options\u001b[0m\n",
      "  \u001b[0;34mtrace_report_suffix: \u001b[0;32m2025-10-09_15-03-42\u001b[0m\n",
      "\n",
      "\u001b[1mCore Nextflow options\u001b[0m\n",
      "  \u001b[0;34mrunName            : \u001b[0;32mnostalgic_swanson\u001b[0m\n",
      "  \u001b[0;34mcontainerEngine    : \u001b[0;32mdocker\u001b[0m\n",
      "  \u001b[0;34mlaunchDir          : \u001b[0;32m/home/chrissi/NahnsenPrak/nf-core-leoniechrissi\u001b[0m\n",
      "  \u001b[0;34mworkDir            : \u001b[0;32m/home/chrissi/NahnsenPrak/nf-core-leoniechrissi/work\u001b[0m\n",
      "  \u001b[0;34mprojectDir         : \u001b[0;32m/home/chrissi/NahnsenPrak/nf-core-leoniechrissi\u001b[0m\n",
      "  \u001b[0;34muserName           : \u001b[0;32mchrissi\u001b[0m\n",
      "  \u001b[0;34mprofile            : \u001b[0;32mgitpod,docker\u001b[0m\n",
      "  \u001b[0;34mconfigFiles        : \u001b[0;32m/home/chrissi/NahnsenPrak/nf-core-leoniechrissi/nextflow.config\u001b[0m\n",
      "\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\n",
      "-\u001b[2m----------------------------------------------------\u001b[0m-\n",
      "* The nf-core framework\n",
      "    https://doi.org/10.1038/s41587-020-0439-x\n",
      "\n",
      "* Software dependencies\n",
      "    https://github.com/nf-core/leoniechrissi/blob/master/CITATIONS.md\n",
      "\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…RISSI:LEONIECHRISSI:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…RISSI:LEONIECHRISSI:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[4A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…RISSI:LEONIECHRISSI:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[7A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…RISSI:LEONIECHRISSI:\u001b[mFASTQC -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "\u001b[8A\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…RISSI:LEONIECHRISSI:\u001b[mFASTQC\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "\u001b[8A\n",
      "\u001b[2mexecutor >  local (1)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "\u001b[9A\n",
      "\u001b[2mexecutor >  local (1)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "\u001b[9A\n",
      "\u001b[2mexecutor >  local (1)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…I:LEONIECHRISSI:\u001b[mTRIMGALORE\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "\u001b[12A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "\u001b[12A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "\u001b[12A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf\u001b[K\n",
      "\u001b[13A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  [ ! -f  SAMPLE_SINGLE_END.fastq.gz ] && ln -s Small_SRR19383412_1.fastq.gz Small_SRR19383412_2.fastq.gz SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  trim_galore \\\u001b[K\n",
      "       \\\u001b[K\n",
      "      --cores 1 \\\u001b[K\n",
      "      --gzip \\\u001b[K\n",
      "      SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE\":\u001b[K\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001b[K\n",
      "      cutadapt: $(cutadapt --version)\u001b[K\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  ln: target 'SAMPLE_SINGLE_END.fastq.gz': No such file or directory\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /home/chrissi/NahnsenPrak/nf-core-leoniechrissi/work/9f/d9cbc7fec85a7f11983a594282282b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[53A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[31m ✘\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC -\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  [ ! -f  SAMPLE_SINGLE_END.fastq.gz ] && ln -s Small_SRR19383412_1.fastq.gz Small_SRR19383412_2.fastq.gz SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  trim_galore \\\u001b[K\n",
      "       \\\u001b[K\n",
      "      --cores 1 \\\u001b[K\n",
      "      --gzip \\\u001b[K\n",
      "      SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE\":\u001b[K\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001b[K\n",
      "      cutadapt: $(cutadapt --version)\u001b[K\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  ln: target 'SAMPLE_SINGLE_END.fastq.gz': No such file or directory\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /home/chrissi/NahnsenPrak/nf-core-leoniechrissi/work/9f/d9cbc7fec85a7f11983a594282282b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[57A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[31m ✘\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "-\u001b[0;35m[nf-core/leoniechrissi]\u001b[0;31m Pipeline completed with errors\u001b[0m-\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  [ ! -f  SAMPLE_SINGLE_END.fastq.gz ] && ln -s Small_SRR19383412_1.fastq.gz Small_SRR19383412_2.fastq.gz SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  trim_galore \\\u001b[K\n",
      "       \\\u001b[K\n",
      "      --cores 1 \\\u001b[K\n",
      "      --gzip \\\u001b[K\n",
      "      SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE\":\u001b[K\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001b[K\n",
      "      cutadapt: $(cutadapt --version)\u001b[K\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  ln: target 'SAMPLE_SINGLE_END.fastq.gz': No such file or directory\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /home/chrissi/NahnsenPrak/nf-core-leoniechrissi/work/9f/d9cbc7fec85a7f11983a594282282b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[31m ✘\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "-\u001b[0;35m[nf-core/leoniechrissi]\u001b[0;31m Pipeline completed with errors\u001b[0m-\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  [ ! -f  SAMPLE_SINGLE_END.fastq.gz ] && ln -s Small_SRR19383412_1.fastq.gz Small_SRR19383412_2.fastq.gz SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  trim_galore \\\u001b[K\n",
      "       \\\u001b[K\n",
      "      --cores 1 \\\u001b[K\n",
      "      --gzip \\\u001b[K\n",
      "      SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE\":\u001b[K\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001b[K\n",
      "      cutadapt: $(cutadapt --version)\u001b[K\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  ln: target 'SAMPLE_SINGLE_END.fastq.gz': No such file or directory\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /home/chrissi/NahnsenPrak/nf-core-leoniechrissi/work/9f/d9cbc7fec85a7f11983a594282282b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[58A\n",
      "\u001b[2mexecutor >  local (2)\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m5c/4d44a6\u001b[0;2m] \u001b[0;2m\u001b[mNFC…FASTQC\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 1 of 1\u001b[32m ✔\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m9f/d9cbc7\u001b[0;2m] \u001b[0;2m\u001b[mNFC…GALORE\u001b[33;2m (\u001b[0;33mSAMPLE_SINGLE_END\u001b[2m)\u001b[m\u001b[2m |\u001b[m 0 of 1\u001b[31m ✘\u001b[m\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…:\u001b[mHISAT2_EXTRACTSPLICESITES\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_BUILD -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…LEONIECHRISSI:\u001b[mHISAT2_ALIGN -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:\u001b[mSUBREAD_FEATURECOUNTS -\u001b[K\n",
      "\u001b[2m[\u001b[0;34m-        \u001b[0;2m] \u001b[0;2mNFC…ISSI:LEONIECHRISSI:\u001b[mMULTIQC\u001b[2m |\u001b[m 0 of 1\u001b[K\n",
      "[DUMP: gtf3] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf2] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "[DUMP: gtf1] [['id':'genes'], /ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf]\u001b[K\n",
      "Staging foreign file: s3://ngi-igenomes/igenomes/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf\u001b[K\n",
      "Execution cancelled -- Finishing pending tasks before exit\u001b[K\n",
      "-\u001b[0;35m[nf-core/leoniechrissi]\u001b[0;31m Pipeline completed with errors\u001b[0m-\u001b[K\n",
      "\u001b[31mERROR ~ Error executing process > 'NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)'\u001b[K\n",
      "\u001b[K\n",
      "Caused by:\u001b[K\n",
      "  Process `NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE (SAMPLE_SINGLE_END)` terminated with an error exit status (1)\u001b[K\n",
      "\u001b[K\n",
      "\u001b[K\n",
      "Command executed:\u001b[K\n",
      "\u001b[K\n",
      "  [ ! -f  SAMPLE_SINGLE_END.fastq.gz ] && ln -s Small_SRR19383412_1.fastq.gz Small_SRR19383412_2.fastq.gz SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  trim_galore \\\u001b[K\n",
      "       \\\u001b[K\n",
      "      --cores 1 \\\u001b[K\n",
      "      --gzip \\\u001b[K\n",
      "      SAMPLE_SINGLE_END.fastq.gz\u001b[K\n",
      "  \u001b[K\n",
      "  cat <<-END_VERSIONS > versions.yml\u001b[K\n",
      "  \"NFCORE_LEONIECHRISSI:LEONIECHRISSI:TRIMGALORE\":\u001b[K\n",
      "      trimgalore: $(echo $(trim_galore --version 2>&1) | sed 's/^.*version //; s/Last.*$//')\u001b[K\n",
      "      cutadapt: $(cutadapt --version)\u001b[K\n",
      "      pigz: $( pigz --version 2>&1 | sed 's/pigz //g' )\u001b[K\n",
      "  END_VERSIONS\u001b[K\n",
      "\u001b[K\n",
      "Command exit status:\u001b[K\n",
      "  1\u001b[K\n",
      "\u001b[K\n",
      "Command output:\u001b[K\n",
      "  (empty)\u001b[K\n",
      "\u001b[K\n",
      "Command error:\u001b[K\n",
      "  ln: target 'SAMPLE_SINGLE_END.fastq.gz': No such file or directory\u001b[K\n",
      "\u001b[K\n",
      "Work dir:\u001b[K\n",
      "  /home/chrissi/NahnsenPrak/nf-core-leoniechrissi/work/9f/d9cbc7fec85a7f11983a594282282b\u001b[K\n",
      "\u001b[K\n",
      "Container:\u001b[K\n",
      "  community.wave.seqera.io/library/cutadapt_trim-galore_pigz:a98edd405b34582d\u001b[K\n",
      "\u001b[K\n",
      "Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\u001b[31mERROR ~ Pipeline failed. Please refer to troubleshooting docs: https://nf-co.re/docs/usage/troubleshooting\u001b[K\n",
      "\u001b[K\n",
      " -- Check '.nextflow.log' file for details\u001b[39m\u001b[K\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Beispielaufruf der nf-core leoniechrissi pipeline mit unserem  adjusted samplesheet\n",
    "!nextflow run ./main.nf \\\n",
    "    --input ./assets/samplesheet_copy.csv \\\n",
    "    --outdir results/ \\\n",
    "    --genome R64-1-1 \\\n",
    "    -profile gitpod,docker \\\n",
    "    -process.memory '8.GB' \\\n",
    "    -dump-channels \\\n",
    "    -process.cpus 2 -resume"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "workflows",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
